File: src/main/java/alankstewart/math/lines/TwoThousandLines.java
package alankstewart.math.lines;

import java.awt.*;
import java.awt.geom.Line2D;
import java.awt.geom.Point2D;

import static java.lang.Math.PI;
import static java.lang.Math.cos;
import static java.lang.Math.pow;
import static java.lang.Math.sin;

/**
 * Created by alankstewart on 23/12/14.
 */
public final class TwoThousandLines implements Lines {

    private static final int SIZE = 1024;

    public static void main(final String[] args) {
        final int size = args.length > 0 && args[0] != null && args[0].matches("-?\\d+") ? Integer.parseInt(args[0]) : SIZE;
        final TwoThousandLines twoThousandLines = new TwoThousandLines();
        twoThousandLines.display(size);
    }

    @Override
    public LineComponent getLineComponent(final Dimension d) {
        return new TwoThousandLinesComponent(d);
    }

    static class TwoThousandLinesComponent extends LineComponent {

        private static final int LINES = 2000;

        public TwoThousandLinesComponent(final Dimension d) {
            super(d);
        }

        @Override
        protected int getLines() {
            return LINES;
        }

        @Override
        protected Line2D.Double getLineSegment(final int n) {
            return new Line2D.Double(getPoint(12, 10, n), getPoint(8, 6, n));
        }

        private Point2D.Double getPoint(final int l, final int m, final int n) {
            return new Point2D.Double(pow(sin(l * PI * n / LINES), 3) * tx, pow(cos(m * PI * n / LINES), 3) * tx);
        }
    }
}

File: src/main/scala/alankstewart/math/lineart/TwoThousandLines.scala
package alankstewart.math.lineart

import java.awt.geom.{Line2D, Point2D}

import scala.math.{Pi, cos, pow, sin}
import scala.swing._

/**
  * Created by alanstewart on 22/02/2016.
  */
object TwoThousandLines extends App with Lines {

  val Size = 1024

  val size = if (args.length > 0 && args(0).matches("-?\\d+")) args(0).toInt else Size
  display(size)

  override def getLineComponent(d: Dimension) = {
    new TwoThousandLinesComponent(d)
  }

  class TwoThousandLinesComponent(d: Dimension) extends LineComponent(d) {

    val Lines = 2000

    override def getLines = {
      Lines
    }

    override def getLineSegment(n: Int) = {
      new Line2D.Double(getPoint(12, 10, n), getPoint(8, 6, n))
    }

    private def getPoint(l: Int, m: Int, n: Int) = {
      new Point2D.Double(pow(sin(l * Pi * n / Lines), 3) * tx, pow(cos(m * Pi * n / Lines), 3) * tx)
    }
  }

}
